components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/AuthenticationContextCache.java

```diff
@@ -126,6 +151,17 @@ public AuthenticationContextCacheEntry getValueFromCache(AuthenticationContextCa
             // Update the cache again with the new value.
             super.addToCache(key, entry);
         }
+        if (isSessionDataStorageOptimizationEnabled) {
+            if (entry != null) {
+                try {
+                    AuthenticationContextLoader.getInstance().loadAuthenticationContext(entry.getContext());
+                } catch (AuthenticationContextLoaderException e) {
+                    log.debug(String.format("Error occurred while loading optimized authentication context with " +
+                            "context id: %s", entry.getContext().getContextIdentifier()), e);
+                    entry = null;
```

Comment:
what's the purpose of setting it to null here?

Reply:
If there is a possibility of exception, it would have occurred when loading the object using the reference. In that case the context will not be a complete, therefore we return null. Framework will handle this condition as null entry.